src/puppetlabs/puppetdb/command.clj

```diff
@@ -315,24 +316,35 @@
     containing :received-commands (a count of the commands received so
     far by the current service instance), and :executed-commands (a
     count of the commands that the current instance has processed
-    without triggering an exception)."))
+    without triggering an exception).")
+
+  (response-pub [this]
+    "Returns a core.async publishing channel to which messages are written after
+     they have been processed. The topic for each message is its uuid." ))
 
 (defservice command-service
   PuppetDBCommandDispatcher
   [[:PuppetDBServer shared-globals]
    [:MessageListenerService register-listener]]
   (init [this context]
-    (assoc context :stats (atom {:received-commands 0
-                                 :executed-commands 0})))
+    (let [response-pub-chan (async/chan)]
+      (assoc context
+             :stats (atom {:received-commands 0
+                           :executed-commands 0})
+             :response-pub-chan response-pub-chan
+             :response-pub (async/pub response-pub-chan #(-> % :annotations :id str)))))
+
   (start [this context]
     (let [{:keys [scf-write-db catalog-hash-debug-dir]} (shared-globals)
           config {:db scf-write-db
-                  :catalog-hash-debug-dir catalog-hash-debug-dir}]
+                  :catalog-hash-debug-dir catalog-hash-debug-dir}
+          {:keys [response-pub-chan response-pub]} context]
       (register-listener
        supported-command?
        (fn [cmd]
          (let [result (process-command! cmd config)]
            (swap! (:stats context) update :executed-commands inc)
+           (async/>!! response-pub-chan cmd)
```

Comment:
Yeah, from the docs:
"Items received when there are no matching subs get dropped."